tools/RAiDER/cli/__main__.py

```diff
@@ -0,0 +1,455 @@
+    geographic bounding box :
+    downloadGNSSdelay.py --download --out products -y 20100101 20141231 --returntime '00:00:00' -b '39 40 -79 -78'
+    """)
+
+    # Stations to check/download
+    area = p.add_argument_group(
+        'Stations to check/download. Can be a lat/lon bounding box or file, or will run the whole world if not specified')
+    area.add_argument(
+        '--station_file', '-f', default=None, dest='station_file',
+        help=('Text file containing a list of 4-char station IDs separated by newlines'))
+    area.add_argument(
+        '-b', '--bounding_box', dest='bounding_box', type=str, default=None,
+        help="Provide either valid shapefile or Lat/Lon Bounding SNWE. -- Example : '19 20 -99.5 -98.5'")
+    area.add_argument(
+        '--gpsrepo', '-gr', default='UNR', dest='gps_repo',
+        help=('Specify GPS repository you wish to query. Currently supported archives: UNR.'))
+
+    misc = p.add_argument_group("Run parameters")
+    add_out(misc)
+
+    misc.add_argument(
+        '--date', dest='dateList',
+        help=dedent("""\
+            Date to calculate delay.
+            Can be a single date, a list of two dates (earlier, later) with 1-day interval, or a list of two dates and interval in days (earlier, later, interval).
+            Example accepted formats:
+               YYYYMMDD or
+               YYYYMMDD YYYYMMDD
+               YYYYMMDD YYYYMMDD N
+            """),
+        nargs="+",
+        action=DateListAction,
+        type=date_type,
+        required=True
+    )
+
+    misc.add_argument(
+        '--returntime', dest='returnTime',
+        help="Return delays closest to this specified time. If not specified, the GPS delays for all times will be returned. Input in 'HH:MM:SS', e.g. '16:00:00'",
+        default=None)
+
+    misc.add_argument(
+        '--download',
+        help='Physically download data. Note this option is not necessary to proceed with statistical analyses, as data can be handled virtually in the program.',
+        action='store_true', dest='download', default=False)
+
+    add_cpus(misc)
+    add_verbose(misc)
+
+    args =  p.parse_args()
+
+    dlGNSS(args)
+    return
+
+
+## ------------------------------------------------------------ prepFromARIA.py
+def calcDelaysARIA(iargs=None):
+    from RAiDER.aria.prepFromARIA import main as ARIA_prep
+    from RAiDER.aria.calcARIA    import main as ARIA_calc
+
+    p = argparse.ArgumentParser(
+        description='Prepare files from ARIA-tools output to use with RAiDER')
+
+    p.add_argument(
+        'file', type=str,
+        help='1 ARIA GUNW netcdf file'
+        )
+
+    p.add_argument(
+        '-m', '--model', default='ERA5', type=str,
+        help='Weather model (default=HRRR)'
+        )
+
+    p.add_argument(
+        '-d', '--dem', default='', type=str,
+        help='Custom DEM. Default: Copernicus GLO30 (downloaded on fly)'
+        )
+
```

Comment:
technically no DEM is needed for the 3D cube to be made.

Reply:
This goes back to my question about primary product for GUNW.

Reply:
We make a 3D cube of the delays. A DEM is not needed for that; just height levels and lateral spacing.